tools/docker/corda-all-in-one/corda-v5/cordapps-deployment/healthcheck.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+# Check if CPI has been generated
+check_cpi_exists() {
+  local file_path="/CSDE-cordapp-template-kotlin/workflows/build/MyCorDapp-1.0-SNAPSHOT.cpi"
+
+  if [ ! -f "$file_path" ]; then
+    echo "CPI file not found at $file_path. Exiting with status code 1."
+    exit 1
+  else
+    echo "CPI file found at $file_path."
+  fi
+}
+
+# Check if the endpoint is accessible
+checkCurlSuccess() {
+    echo "Executing checkCurlSuccess function..."
+    local response_code
+    response_code=$(curl -k -s -o /dev/null -w "%{http_code}" -u admin:admin "https://localhost:8888/api/v1/cpi")
+    if [ "$response_code" -eq 200 ]; then
+        echo "CURL request successful."
+    else
+        echo "CURL request failed with response code: $response_code"
+        exit 1
+    fi
+}
+
+
+check_cpi_exists
+
+checkCurlSuccess
+
+WAIT_TIME=30
```

Comment:
What is this wait time for? We need to have some logic to remove the need for this (As variable environments will have very different wait-time to run this)

Reply:
If it's not mention properly in corda docs, you can create an issue on their github repo having this, while also creating an issue on our board referencing that. (Someone else from the team can pick it up from there once the corda team responds as I know you are working on corda for a long time and we can enable others to pickup from here.)

Reply:
I ran the test multiple times and opted to remove the wait time because it is not needed for this scenario. The existing health check is already enough.